Enhancements to sda-admin

andkaha opened the pull request:
This PR closes https://github.com/NBISweden/LocalEGA-SE-Deployment/issues/641

This PR adds automatic dumping of the RabbitMQ queues as JSON files. Before a queue is modified by the `sda-admin` script, it is backed up as `<queue>-0.json` (any existing file with that name is rotated up to `<queue>-1.json`, etc., up to `<queue>-9.json`).

The PR also adds a `restore` sub-command to the `sda-admin` script with the following documentation:
```text
The "restore" sub-command is used for restoring the RabbitMQ
messages from one of the "<queue>-<n>.json" files that are
automatically created by this script.  The messages in the JSON
file are added to the messages in the corresponding RabbitMQ
queue, but if there are duplicates based on the filepath in
the messages' payload, the duplicate(s) from the queue will be
dropped.

Example usage:

    Restoring the messages from the "inbox-0.json" file (the
    most recent backup of the "inbox" queue):

    ./sda-admin restore inbox-0.json
```


The PR also contains t a refactoring of `certs/make_certs.sh` for the sake of readability, adding comments and simplifying some of the code.

andkaha committed:
Also mention the `S3_ACCESS_KEY` environment variable

andkaha committed:
1st pass

* Use the `bash` shell.
* Use real arrays.
* Add newlines for readability.

andkaha committed:
2nd pass

* Read the serial number from a file in the script directory (and then
  increment it and store it again).
* Remove references to `$out_dir` by changing to that directory at the
  start.

andkaha committed:
Do serial numbers properly and use `realpath` to figure out script directory

andkaha committed:
Rewrite for `sh`

andkaha committed:
typo

andkaha committed:
Tighter

andkaha committed:
A tiny bit of documentation

andkaha committed:
use the `--json` shortcut with `curl`

andkaha committed:
Save queue JSON dumps

A backup of the queue `<queue>` will be saved to `<queue>-0.json`, and
existing such files will be rotated up to `<queue>-9.json`.

andkaha committed:
Don't use base64-encoded payloads

This makes it easier to read the message queue dumps.

andkaha committed:
Merge branch 'main' into todo/sda-admin-updates

andkaha committed:
Merge branch 'bug/fix-demo-compose' into todo/sda-admin-updates

andkaha committed:
Smaller comment fixes

andkaha committed:
Move `do` to new line due to long list

andkaha committed:
Merge branch 'todo/cert-script-cleanup' into todo/sda-admin-updates

andkaha committed:
Call it `$dumpfile` rather than `tmpfile`

andkaha committed:
Implement `restore` to restore messages from file

andkaha committed:
Document `restore`

andkaha committed:
Grammar

andkaha committed:
Merge branch 'main' into todo/sda-admin-updates

andkaha committed:
ShellCheck

andkaha committed:
Forgot `-r` (no functional difference here, but only by luck)

andkaha committed:
Only call `curl` to get all messages if previous `cat` succeeded

norling reviewed (approved):
Looks really good, as usual! :)

andkaha committed:
Incorporate suggestions from Kostas

The rather than using "gdi/sda" as the exchange prefix string, use only
"sda" and prefix *that* with the queue prefix ("gdi" by default).

andkaha committed:
Comments, mostly (and check that dump file exists in `restore`)

kostas-kou reviewed (approved):
Looks great. We have discussed my questions on Friday
